lib/home/profile/account/change_password.dart

```diff
@@ -1,17 +1,219 @@
   const ChangePasswordScreen({super.key});
 
+  @override
+  // ignore: library_private_types_in_public_api
+  _ChangePasswordScreenState createState() => _ChangePasswordScreenState();
+}
+
+class _ChangePasswordScreenState extends State<ChangePasswordScreen> {
+  final TextEditingController _currentPasswordController =
+      TextEditingController();
+  final TextEditingController _newPasswordController = TextEditingController();
+  final TextEditingController _confirmPasswordController =
+      TextEditingController();
+  bool _isCurrentPasswordCorrect = false;
+  bool _isCurrentPasswordVerified = false;
+  String _verificationMessage = '';
+  Color _messageColor = Colors.black;
+  String _passwordError = '';
+  bool _isPasswordValid = false;
+
+  @override
+  void dispose() {
+    _currentPasswordController.dispose();
+    _newPasswordController.dispose();
+    _confirmPasswordController.dispose();
+    super.dispose();
+  }
+
+  void _validateCurrentPassword() async {
+    final user = FirebaseAuth.instance.currentUser;
+    final cred = EmailAuthProvider.credential(
+      email: user!.email!,
+      password: _currentPasswordController.text,
+    );
+
+    try {
+      await user.reauthenticateWithCredential(cred);
+      setState(() {
+        _isCurrentPasswordCorrect = true;
+        _isCurrentPasswordVerified = true;
+        _verificationMessage = 'Verification successful';
+        _messageColor = Colors.green;
+      });
+    } catch (e) {
+      setState(() {
+        _isCurrentPasswordCorrect = false;
+        _isCurrentPasswordVerified = true;
+        _verificationMessage = 'Verification failed';
+        _messageColor = Colors.red;
+      });
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text('Change Password'),
       ),
-      body: const Center(
-        child: Text('Change password page put here'),
+      body: Center(
+        child: Container(
+          padding: const EdgeInsets.all(16.0),
+          width: 500,
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              TextField(
+                controller: _currentPasswordController,
+                obscureText: true,
+                decoration: const InputDecoration(
+                  labelText: 'Current Password',
+                  labelStyle: TextStyle(color: Colors.black),
+                ),
+              ),
+              const SizedBox(height: 8),
+              ElevatedButton(
+                onPressed: _validateCurrentPassword,
+                child: const Text('Verify Current Password'),
+              ),
+              if (_isCurrentPasswordVerified)
+                Text(
+                  _verificationMessage,
+                  style: TextStyle(
+                      color: _messageColor, fontWeight: FontWeight.bold),
```

Comment:
```suggestion
                  style: StyleText(color: _messageColor, weight: 700),
```

The [`StyleText`](https://github.com/theheartcenter-inc/thc/blob/e24a917a5bd1a4bc04c27847a1f85fc40e54f0aa/lib/utils/style_text.dart#L8) class helps us maintain a consistent font family across the app, and it allows for a variable font weight (any number between 50 and 1000!)